coulomb/js/src/main/scala/coulomb/typename.scala

```diff
@@ -0,0 +1,54 @@
+package coulomb
+
+import scala.language.experimental.macros
+
+import scala.reflect.macros.whitebox
+
+import scala.reflect.macros.blackbox.Context
+
+trait UnitTypeName[T] {
+  def name: String // typeName
+  def typeString: String
```

Comment:
is there a reason for having both `name` and `typeString` ?

Reply:
Yes, the implementation on the jvm side using `WeakTagTypes` differs notably and I wanted to preserve that.